go/ct/gen/state_test.go

```diff
@@ -478,3 +506,149 @@ func TestStateGenerator_BlockNumberHashes(t *testing.T) {
 		newHashes = append(newHashes, hashi)
 	}
 }
+
+// //////////////////////////////////////////////////////////
+// code constraints
+func TestStateGenerator_AddCodeOperation(t *testing.T) {
+	generator := NewStateGenerator()
+	var1 := Variable("a")
+	generator.AddCodeOperation(var1, vm.ADD)
+	varOpConstraint := varOpConstraint{var1, vm.ADD}
+	if len(generator.codeGen.varOps) != 1 || generator.codeGen.varOps[0] != varOpConstraint {
```

Comment:
I wanted to keep the tests as minimal and unitarian as possible, since Generate is such a complex function and it has a lot of other tests addressing it, I thought here aiming for just the explicit effect of the function being tested is enough.